rust/agents/scraper/src/scraper/mod.rs

```diff
@@ -49,143 +48,231 @@ impl BaseAgent for Scraper {
+                locals.insert(outbox_domain, local);
+            }
+
+            for (_, inbox_config) in chain_config.inboxes.iter() {
+                if let Some(remote) =
+                    Self::load_inbox(&chain_config, inbox_config, &metrics).await?
+                {
+                    let inbox_remote_domain = remote.inbox.remote_domain();
+                    let inbox_local_domain = remote.inbox.local_domain();
+                    assert_eq!(inbox_remote_domain, outbox_domain);
+                    assert_ne!(
+                        inbox_local_domain, outbox_domain,
+                        "Attempting to load inbox for the chain we are on"
+                    );
+
+                    trace!(
+                        local_domain = inbox_local_domain,
+                        remote_domain = inbox_remote_domain,
+                        "Created inbox and inbox indexer"
+                    );
+                    remotes
+                        .entry(inbox_local_domain)
+                        .or_default()
+                        .insert(inbox_remote_domain, remote);
+                }
+            }
+
+            index_settings.insert(outbox_domain, chain_config.index);
+        }
+
+        let contract_sync_metrics = ContractSyncMetrics::new(metrics.clone());
+        let mut scrapers: HashMap<u32, SqlChainScraper> = HashMap::new();
+        for (local_domain, local) in locals.into_iter() {
+            let remotes = remotes.remove(&local_domain).unwrap_or_default();
+            let index_settings = index_settings
+                .remove(&local_domain)
+                .expect("Missing index settings for domain");
+
+            let scraper = SqlChainScraper::new(
+                db.clone(),
+                local,
+                remotes,
+                &index_settings,
+                contract_sync_metrics.clone(),
+            )
+            .await?;
+            scrapers.insert(local_domain, scraper);
+        }
+
+        trace!(domain_count = scrapers.len(), "Creating scraper");
+
         Ok(Self {
+            db,
             metrics,
-            outboxes,
-            inboxes,
-            gas_paymasters,
+            scrapers,
         })
     }
 
     #[allow(clippy::async_yields_async)]
     async fn run(&self) -> Instrumented<JoinHandle<Result<()>>> {
         let tasks = self
-            .outboxes
+            .scrapers
             .iter()
-            .map(|(name, outbox)| {
-                let span = info_span!("OutboxContractSync", %name, self = ?outbox);
-                let syncer = outbox.clone().sync();
+            .map(|(name, scraper)| {
+                let span = info_span!("ChainContractSync", %name, self = ?scraper);
+                let syncer = scraper.clone().sync();
                 tokio::spawn(syncer).instrument(span)
             })
+            .chain(
+                [tokio::spawn(delivered_message_linker(self.db.clone()))
```

Comment:
TBH, I kinda want to leave it for now since we have already flip-flopped a couple of times. We can remove it during refactoring if it makes sense to. Will add a todo to decide.